src/feedback_vertex_set.c

```diff
@@ -0,0 +1,90 @@
+/* -*- mode: C -*-  */
+/* vim:set ts=4 sw=4 sts=4 et: */
+/*
+   IGraph library.
+   Copyright (C) 2011-2012  Gabor Csardi <csardi.gabor@gmail.com>
+   334 Harvard street, Cambridge, MA 02139 USA
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301 USA
+
+*/
+
+#include "igraph_centrality.h"
+#include "igraph_components.h"
+#include "igraph_constants.h"
+#include "igraph_datatype.h"
+#include "igraph_dqueue.h"
+#include "igraph_error.h"
+#include "igraph_glpk_support.h"
+#include "igraph_interface.h"
+#include "igraph_memory.h"
+#include "igraph_structural.h"
+#include "igraph_types.h"
+#include "igraph_visitor.h"
+
+/**
+ * \ingroup structural
+ * \function igraph_feedback_vertex_set
+ * \brief Calculates a feedback vertex set of the graph.
+ *
+ * </para><para>
+ * A feedback vertex set is a set of vertices whose removal makes the graph acyclic.
+ *
+ * </para><para>
+ * The problem is NP complete except in undirected graphs of maximum degree three.
```

Comment:
The *minimum* feedback vertex set is NP-complete. Finding an arbitrary feedback vertex set (FVS) is not.

I assume this function will use a heuristic to find a "reasonably small" FVS, but not a minimum one because it would be counterproductive to use an exponential complexity algorithm as a subroutine in a polynomial-time one. 

Can you please clarify these in the description?

Reply:
I changed the name appending `_approx` and clarified a little in the function doc. Implementation of a decent algorithm is still missing, returning all vertices for now.